src/map.cpp

```diff
@@ -7334,6 +7334,13 @@ void map::spawn_monsters_submap( const tripoint &gp, bool ignore_sight )
             if( i.friendly ) {
                 tmp.friendly = -1;
             }
+            if( !i.data.ammo.empty() ) {
+                for( std::pair<std::string, jmapgen_int> ap : i.data.ammo ) {
+                    tmp.ammo.emplace( ap.first, ap.second.get() );
```

Comment:
Possibly broke compiling on Jenkins (see https://ci.narc.ro/job/Cataclysm-Matrix/10700/Graphics=Curses,Platform=Linux_x64/console).

Reply:
`ap.first` should be probably wrapped into `itype_id`:

![image](https://user-images.githubusercontent.com/16213433/83000771-81c1b580-a013-11ea-8ffc-0260b7607f49.png)

Reply:
Or maybe `spawn_data` struct should be changed and use `itype_id` instead of a string:

```CPP
struct spawn_data {
    std::map<itype_id, jmapgen_int> ammo;

    spawn_data() : ammo() {};
};
```

Reply:
And all of that happened because `https://github.com/CleverRaven/Cataclysm-DDA/pull/40637` that changed `itype_id` from a simple string typedef to a proper type was created after current PR, but merged prior to that PR.